docs/source/metrics.rst

```diff
@@ -4,7 +4,16 @@ ignite.metrics
 Metrics provide a way to compute various quantities of interest in an online
 fashion without having to store the entire output history of a model.
 
-In practice a user needs to attach the metric instance to an engine. The metric
+Attach Engine API
+------------------
+
+The metrics as stated above are computed in a online fashion,which means that the metric instance accumulates some internal counter on 
+each iteration and accuracy value is computed once epoch is ended. Internal counters are reset after every epoch.
+For doing this we use three public APIs ``reset`` , ``update`` and ``compute`` :ref:`read about them here<Reset, Update, Compute API>`. We internally accumulate some counters
+when there is a update call. The metric's value is computed on each compute call and counters are reset on each reset
+call. 
+
+In practice a user needs to attach the metric instance to an engine which attaches the ``reset`` , ``compute`` and ``update`` methods to the events. The metric
```

Comment:
> methods to the events

This part can be explained more. Which events and which handler.

Reply:
Thanks for working on that. Now the text provides too much unrelated info. We do not need to explain what are events in the section about metrics. We can provide a link on resources.

> Which events and which handler.

This means explain to which event method `reset` is attached and why. Same for update and compute.

Reply:
@vfdev-5 I actually don't know correctly, I checked the docs but couldn't find out as well. But as reset() is called at the start of the epoch so is it attached to the EPOCH_STARTED event ?